libcaf_core/caf/typed_actor.hpp

```diff
@@ -55,30 +57,34 @@ class typed_actor : detail::comparable<typed_actor<Sigs...>>,
   static constexpr bool has_weak_ptr_semantics = false;
 
   /// Stores the template parameter pack.
-  using signatures = type_list<Sigs...>;
+  using trait = detail::to_trait_t<TraitOrSignature>;
+
+  /// Stores the template parameter pack.
+  using signatures = typename trait::signatures;
```

Comment:
Push this above the static assert to avoid the repeated `to_trait_t<TraitOrSignature>`.